packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -554,11 +556,12 @@ class IOSSimulator extends Device {
   String get sdkNameAndVersion => category;
 
   @override
-  DeviceLogReader get logReader {
-    if (_logReader == null)
-      _logReader = new _IOSSimulatorLogReader(this);
+  DeviceLogReader get logReader => logReaderForApp(null);
```

Comment:
Yep. My thinking is that if `app` can be null maybe it should be an optional param, but then that seems unexpected given the method name. Alternatively, you could document the behaviour when it's null. Nicest might be to rename the method `getLogReader({ApplicationPackage: app})` and kill off the getter altogether. 

As I say, I don't have particularly strong feelings on it; mostly just a nitpick.